src/bin/sid.rs

```diff
@@ -4,40 +4,104 @@ use sid::*;
 
 use clap::Parser;
 
+#[cfg(feature= "debug-gui")]
```

Comment:
You would probably want to create another binary for this purpose, rather than making the same binary different depending on features.

Just write it as a separate file named something like `debug.rs` in the same folder as `sid.rs` and add a `[[bin]]` entry to the `Cargo.toml`. Or if you don't wish to look into that I can do it after merging.

Reply:
Oh won't the whole crate still have all the dependencies in that case? Or maybe that's fine?

I figured the "real" solution was making a seperate crate which used this crates public API.
I was just not feeling like making another repo atm

But sure go wild

Reply:
You can configure it so that building that binary has the dependency, but otherwise it isn't needed. I have set up a similar thing for the "classic UI" binary for my add-ed library crate.